packages/common/src/types/blockItemSummary.ts

```diff
@@ -252,3 +256,302 @@ export type BlockItemStatus =
+ *
+ * @param {BlockItemSummary} summary - The block item summary to check.
+ *
+ * @returns {boolean} whether summary is of type `UpdateContractSummary`.
+ */
+export const isUpdateContractSummary = (
+    summary: BlockItemSummary
+): summary is BaseAccountTransactionSummary & UpdateContractSummary =>
+    summary.type === TransactionSummaryType.AccountTransaction &&
+    summary.transactionType === TransactionKindString.Update;
+
+/**
+ * Type predicate for transfer-like transactions.
+ *
+ * @param {BlockItemSummary} summary - The block item summary to check.
+ *
+ * @returns {boolean} whether summary is of type `TransferSummary` or `TransferWithMemoSummary`.
+ */
+export const isTransferLikeSummary = (
+    summary: BlockItemSummary
+): summary is BaseAccountTransactionSummary &
+    (TransferSummary | TransferWithMemoSummary) =>
+    summary.type === TransactionSummaryType.AccountTransaction &&
+    (summary.transactionType === TransactionKindString.Transfer ||
+        summary.transactionType === TransactionKindString.TransferWithMemo);
+
+/**
+ * Type predicate for {@link FailedTransactionSummary}.
+ *
+ * @param {BlockItemSummary} summary - The block item summary to check.
+ *
+ * @returns {boolean} whether transaction was rejected.
+ */
+export const isRejectTransaction = (
+    summary: BlockItemSummary
+): summary is BaseAccountTransactionSummary & FailedTransactionSummary =>
+    summary.type === TransactionSummaryType.AccountTransaction &&
+    summary.transactionType === TransactionKindString.Failed;
+/**
+ * Helper function to determine whether a transaction was successful (inverse of {@link isRejectTransaction}).
+ *
+ * @param {BlockItemSummary} summary - The block item summary to check.
+ *
+ * @returns {boolean} whether transaction was successful.
+ */
+export const isSuccessTransaction = (
+    summary: BlockItemSummary
+): summary is Exclude<BlockItemSummary, FailedTransactionSummary> =>
+    !isRejectTransaction(summary);
+
+/**
+ * Gets the {@link RejectReason} for rejected transction.
+ *
+ * @param {BlockItemSummary} summary - The block item summary to check.
+ *
+ * @returns {RejectReason | undfined} Reject reason if `summary` is a rejected transaction. Otherwise returns undefined.
+ */
+export function getTransactionRejectReason<T extends FailedTransactionSummary>(
+    summary: T
+): RejectReason;
+export function getTransactionRejectReason(
+    summary: AccountCreationSummary | UpdateSummary
+): undefined;
+export function getTransactionRejectReason(
+    summary: Exclude<AccountTransactionSummary, FailedTransactionSummary>
+): undefined;
+export function getTransactionRejectReason(
+    summary: BlockItemSummary
+): RejectReason | undefined;
+export function getTransactionRejectReason(
+    summary: BlockItemSummary
+): RejectReason | undefined {
+    if (!isRejectTransaction(summary)) {
+        return undefined;
+    }
+
+    return summary.rejectReason;
+}
+
+const isEqualContract =
```

Comment:
Move this to `types` and make it public? This might be useful for people, right?